### PR TITLE
User implementation

### PR DESCRIPTION
**Pull request by ADIGrimm**

### CODE REVIEW

File: src/main/java/online/bookstore/model/User.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@SQLRestriction("is_deleted=false")
+@Getter
+@Setter
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
+    @Column(unique = true)
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/User.java]:
```suggestion
    @Column(unique = true, nullable = false)
```

File: src/main/java/online/bookstore/model/User.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    private String email;
+    @NotNull
+    private String password;
+    @NotNull
+    private String firstName;
+    @NotNull
+    private String lastName;
+    private String shippingAddress;
+    @NotNull
+    @Column(columnDefinition = "TINYINT(1)")
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/User.java]: same here
be consistent, don't need to use different validation annotations

File: src/main/java/online/bookstore/model/Book.java [hunk 3]
```diff
@@ -34,6 +34,7 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
-    @Column(nullable = false)
+    @NotNull
+    @Column(columnDefinition = "TINYINT(1)")
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/Book.java]:
```suggestion
    @Column(columnDefinition = "TINYINT(1)", nullable = false)
```

File: src/main/java/online/bookstore/model/Book.java [hunk 4]
```diff
@@ -34,6 +34,7 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
-    @Column(nullable = false)
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/Book.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/online/bookstore/model/User.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
```
Comment (okuzan) [inline src/main/java/online/bookstore/model/User.java]:
```suggestion
@SQLDelete(sql = "UPDATE users SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction("is_deleted = FALSE")
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**